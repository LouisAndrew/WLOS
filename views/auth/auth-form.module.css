.wrapper {
  @apply flex flex-col justify-center h-full;
}

/* .pass-icon {
  fill: theme('colors.primary-yellow');
  @apply absolute bottom-4 right-4 h-6 w-6 cursor-pointer;
} */

.breadcrumb {
  @apply text-shade-gray font-display text-sm tracking-extrawide mt-3;
}

.button {
  @apply w-full justify-center text-base py-3 md:text-sm;
}

.password-toggle {
  @apply absolute bottom-4 right-4 cursor-pointer md:bottom-3 text-primary-yellow duration-200
    opacity-0;
}

.password-toggle.show {
  @apply opacity-100;
}
