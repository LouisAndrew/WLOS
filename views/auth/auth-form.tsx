import { useRouter } from 'next/router'
import React, { FC, useEffect, useState } from 'react'
import { useAuth } from '@h/useAuth'

export type Props = {}

const isSigningIn = (query: object) => query['sign-in'] !== undefined

const AuthForm: FC<Props> = () => {
  const router = useRouter()
  const { signIn, signInWithGoogle, signUp } = useAuth()
  const [authState, setAuthState] = useState<'SIGN_IN' | 'SIGN_UP'>()
  const [formState, setFormState] = useState<{ username: string; password: string }>({
    username: '',
    password: '',
  })

  const headingText = authState === 'SIGN_IN' ? 'Login Now.' : 'Create an account'

  useEffect(() => {
    setAuthState(isSigningIn(router.query) ? 'SIGN_IN' : 'SIGN_UP')
  }, [router])

  const handleClickSignInGoogle = async () => {
    await signInWithGoogle()
  }
  // todo: on success state
  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault()
    const { username, password } = formState
    if (username && password) {
      if (authState === 'SIGN_IN') {
        await signIn(username, password)
      } else {
        await signUp(username, password)
      }
    }
  }

  return (
    <div data-testid="auth-form-wrapper">
      <h1>
        Hey,
        <span className="text-primary-yellow">{headingText}</span>
      </h1>
      <form onSubmit={handleSubmit} role="form">
        <label htmlFor="username">
          Username
          <input
            type="text"
            id="username"
            value={formState.username}
            onChange={(e) => setFormState((prev) => ({ ...prev, username: e.target.value }))}
          />
        </label>
        <label htmlFor="password">
          Password
          <input
            type="text"
            id="password"
            value={formState.password}
            onChange={(e) => setFormState((prev) => ({ ...prev, password: e.target.value }))}
          />
        </label>
        {authState === 'SIGN_IN' ? (
          <>
            <button data-testid="sign-in" type="submit"></button>
            <button data-testid="sign-in-google" onClick={handleClickSignInGoogle}></button>
          </>
        ) : (
          <button data-testid="sign-up" type="submit"></button>
        )}
      </form>
    </div>
  )
}

export default AuthForm
