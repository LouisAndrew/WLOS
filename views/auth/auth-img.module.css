.wrapper {
  @apply relative bg-bg-gray flex flex-center;
}

.background-wrapper {
  @apply absolute z-10 w-full h-full top-0 left-0 px-6
    flex flex-col items-center justify-center;
}

.yellow-stripe {
  transform: skew(-15deg) rotate(-5deg);
  @apply h-24 w-full bg-primary-yellow;
}

.white-stripe {
  transform: skew(-12deg) rotate(-5deg);
  @apply h-28 w-4/5 bg-white mt-8;
}

.img-wrapper {
  @apply max-w-xs relative z-20 w-3/4;
}
