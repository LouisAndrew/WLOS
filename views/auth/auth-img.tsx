import React, { FC } from 'react'
import Image from 'next/image'

const imgSrc = '/auth-athlete-img.png'
import styles from './auth-img.module.css'

export type Props = {}

const AuthImage: FC<Props> = () => {
  return (
    <div data-testid="wrapper" className={styles.wrapper}>
      <div className={styles['background-wrapper']}>
        <div className={styles['yellow-stripe']} />
        <div className={styles['white-stripe']} />
      </div>
      <div className={styles['img-wrapper']}>
        <Image src={imgSrc} width={805} height={1468} layout="responsive" />
      </div>
    </div>
  )
}

export default AuthImage
