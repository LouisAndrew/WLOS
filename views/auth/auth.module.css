.wrapper {
  @apply p-6 bg-bg-gray;
}

.auth-image__wrapper {
  @apply px-12 flex flex-col justify-center h-full;
}

.auth-image__heading {
  letter-spacing: 0.5rem;
  @apply text-base mt-4 italic;
}

.auth-image__next-btn {
  @apply py-0 self-end mt-12;
}
