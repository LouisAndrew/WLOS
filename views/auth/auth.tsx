import React, { FC, useState } from 'react'

import classname from 'classnames/bind'

import styles from './auth.module.css'
import AuthImage from './auth-img'
import AuthForm from './auth-form'

const cx = classname.bind(styles)

export type Props = {}

const Auth: FC<Props> = () => {
  const [displayAuthImage, setDisplayAuthImage] = useState(true)

  const quote = 'Millions of people use this app to make their life better'

  const authFormWrapperClass = cx({
    'auth-form__wrapper': true,
    'display-form': !displayAuthImage,
  })

  return (
    <div data-testid="auth-wrapper" className={`h-full-screen ${styles.wrapper}`}>
      <div className={styles['auth-image__wrapper']}>
        <AuthImage />
        <h5>Get Started</h5>
        <h3 className={styles['auth-image__heading']}>{quote}</h3>
        <button
          className={`btn btn--primary ${styles['auth-image__next-btn']}`}
          onClick={() => setDisplayAuthImage(false)}
        >
          Next
        </button>
      </div>
      <div className={authFormWrapperClass}>
        <AuthForm />
      </div>
    </div>
  )
}

export default Auth
