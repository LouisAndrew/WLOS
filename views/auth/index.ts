import Auth from './auth'
import AuthImage from './auth-img'
import AuthForm from './auth-form'

export { Auth, AuthImage, AuthForm }
