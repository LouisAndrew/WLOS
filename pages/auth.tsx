import { Auth as AuthView } from '../views/auth'

export default function Auth() {
  return <AuthView />
}
