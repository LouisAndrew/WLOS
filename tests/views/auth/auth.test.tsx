import { render, cleanup } from '@testing-library/react'
import { Auth } from '@v/auth'

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: '',
    replace: () => {},
  }),
}))
afterEach(cleanup)

const Component = <Auth />
describe('Auth view component', () => {
  it('renders the component', () => {
    expect(render(Component).queryByTestId('auth-wrapper')).toBeInTheDocument()
  })

  it('matches the snapshot', () => {
    expect(render(Component)).toMatchSnapshot()
  })
})
