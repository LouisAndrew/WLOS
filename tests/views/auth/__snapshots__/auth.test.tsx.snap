// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth view component matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-full-screen wrapper"
        data-testid="auth-wrapper"
      >
        <div
          class="auth-image__wrapper"
        >
          <div
            class="wrapper"
            data-testid="wrapper"
          >
            <div
              class="background-wrapper"
            >
              <div
                class="yellow-stripe"
              />
              <div
                class="white-stripe"
              />
            </div>
            <div
              class="img-wrapper"
            >
              <div
                style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="display: block; box-sizing: border-box; padding-top: 182.36024844720498%;"
                />
                <noscript />
                <img
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </div>
          <h5>
            Get Started
          </h5>
          <h3
            class="auth-image__heading"
          >
            Millions of people use this app to make their life better
          </h3>
          <button
            class="btn btn--primary auth-image__next-btn"
          >
            Next
          </button>
        </div>
        <div
          data-testid="auth-form-wrapper"
        >
          <h1>
            Hey,
            <span
              class="text-primary-yellow"
            >
              Create an account
            </span>
          </h1>
          <form
            role="form"
          >
            <label
              for="username"
            >
              Username
              <input
                id="username"
                type="text"
                value=""
              />
            </label>
            <label
              for="password"
            >
              Password
              <input
                id="password"
                type="text"
                value=""
              />
            </label>
            <button
              data-testid="sign-up"
              type="submit"
            />
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-full-screen wrapper"
      data-testid="auth-wrapper"
    >
      <div
        class="auth-image__wrapper"
      >
        <div
          class="wrapper"
          data-testid="wrapper"
        >
          <div
            class="background-wrapper"
          >
            <div
              class="yellow-stripe"
            />
            <div
              class="white-stripe"
            />
          </div>
          <div
            class="img-wrapper"
          >
            <div
              style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="display: block; box-sizing: border-box; padding-top: 182.36024844720498%;"
              />
              <noscript />
              <img
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
        </div>
        <h5>
          Get Started
        </h5>
        <h3
          class="auth-image__heading"
        >
          Millions of people use this app to make their life better
        </h3>
        <button
          class="btn btn--primary auth-image__next-btn"
        >
          Next
        </button>
      </div>
      <div
        data-testid="auth-form-wrapper"
      >
        <h1>
          Hey,
          <span
            class="text-primary-yellow"
          >
            Create an account
          </span>
        </h1>
        <form
          role="form"
        >
          <label
            for="username"
          >
            Username
            <input
              id="username"
              type="text"
              value=""
            />
          </label>
          <label
            for="password"
          >
            Password
            <input
              id="password"
              type="text"
              value=""
            />
          </label>
          <button
            data-testid="sign-up"
            type="submit"
          />
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
